### ISSUE TITLE
Add a spinner / progress bar when opening the database

### ISSUE THREAD

**Issue by suve** [keepassxreboot/keepassxc#3742]
## Summary
By tweaking the encryption settings for a database, it's possible to reach a decryption time of several seconds. When a database with such strong encryption is opened, the app seemingly freezes - there's nothing that would indicate to the user "hold on, this can take a moment".

## Desired Behavior
When the user inputs their password(s) and presses "OK" to open the database, the "Enter master key" screen is replaced with a spinner or a progress bar.

**Comment by phoerious** [comment]
This is already indicated by a spinning cursor.

**Comment by suve** [comment]
Yes, but I'd argue that's not very visible. When the database has only a password, a user can type the password and press Enter to start the decryption process - so they may not be paying much attention to the cursor.

**Comment by droidmonkey** [comment]
This will require moving the decryption process into a new thread. Right now it is done on the main event thread which locks up the GUI. It's the right thing to do, but introduces interesting issues when saving a database (ie, what happens if I make a change while the database is still saving??). 

For this, I think a progress bar is most appropriate. We can calculate progress by listening in on the number of rounds completed by the KDF. This is what takes the time when decrypting, transforming the key. Now, I'm not sure the argon2 or aes functions have a callback with this information. 

We could always just put up funny messages like "Reticulating Splines" /joke

**Comment by droidmonkey** [comment]
This is related to #2645

**Comment by phoerious** [comment]
IIRC, the decryption process is already in a separate thread to avoid blocking the GUI.

**Comment by ba32107** [comment]
The fix I made for another issue disables the DB open form while the decryption happens. Combined with the mouse wait cursor, I think it's pretty straightforward that the application is doing something.
Ref: https://github.com/keepassxreboot/keepassxc/pull/3885

Not against this idea, but IMO it's not worth the effort.
